src/H5Olayout.c

```diff
@@ -164,29 +180,46 @@ H5O__layout_decode(H5F_t *f, H5O_t H5_ATTR_UNUSED *open_oh, unsigned H5_ATTR_UNU
              * size in the dataset code, where we've got the dataspace
              * information available also.  - QAK 5/26/04
              */
+            if (H5_IS_BUFFER_OVERFLOW(p, (ndims * sizeof(uint32_t)), p_end))
+                HGOTO_ERROR(H5E_OHDR, H5E_OVERFLOW, NULL, "ran off end of input buffer while decoding")
             p += ndims * 4; /* Skip over dimension sizes (32-bit quantities) */
```

Comment:
Could this '4' be replaced with a size call to match the overflow check?

Reply:
Sure. Changed.